IPython/html/static/tree/js/newnotebook.js

```diff
@@ -85,16 +85,20 @@ define([
                     url += "?kernel_name=" + kernel_name;
                 }
                 w.location = url;
-            },
-            function (error) {
-                w.close();
-                dialog.modal({
-                    title : 'Creating Notebook Failed',
-                    body : "The error was: " + error.message,
-                    buttons : {'OK' : {'class' : 'btn-primary'}}
-                });
-            }
-        );
+        }).catch(function (e) {
```

Comment:
Looks like what I said was right.  `P.then(s, e)` leads to an `Unhandled promise rejection Promise`:
http://jsfiddle.net/fq9z10ap/

Whereas `P.then(s).catch(e)` works as intended:
http://jsfiddle.net/4j54cyjy/